src/core/ui.js

```diff
@@ -87,8 +87,55 @@ function errWarn(msg, arr, butName, title) {
 
 function createWarnButton(butName, arr, title) {
   const buttonId = `respec-pill-${butName}`;
+  var numSteps = 20.0;
+  var boxElement
+  var prevRatio = 0.0;
+
+  button.addEventListener("load", function (event) {
+    boxElement = document.querySelector("#respec-pill");
+
+    createObserver();
+  }, false);
+
+  function createObserver() {
+    var observer;
+
+    var options = {
+      root: null,
+      rootMargin: "0px",
+      threshold: buildThresholdList()
+    };
+
+    observer = new IntersectionObserver(handleIntersect, options);
+    observer.observe(boxElement);
+  }
+
+  function buildThresholdList() {
```

Comment:
Can you help me understand what this does? In my mind, all we are concerned about is detecting when `respecUI` intersects a `.logo`.

Reply:
Yeah correct but this help us make a threshold (i.e. he percentage of intersection after which the logo must fade) after which the logo will fade out so unless and until we do not assign a threshold the intersection observer will not do anything

Reply:
ok, but couldn't it just be 1 or 0? (sorry if that's a silly question, I honestly have no idea how it works)

Reply:
Yeah that might be done.. I need to check it out once doing it locally.